log/log.py

```diff
@@ -243,7 +242,7 @@ def set_file(file_path, type='out', level="WARNING"):
                     Log.log.addHandler(status_file_handler)
                 os.chmod(file_path, 509)
             except Exception: # retry again
-                pass
+                sleep(timeout * retries)
```

Comment:
Writing the test for `autosubmit readme` it worked locally, and then failed on GitHub Actions.

That's because I had an `.autosubmitrc` in my local computer, and on GitHub we didn't have one. I used an existing fixture that sets the `CONFIG_FILE_FOUND` but I changed that to `True` (it was `False`, but was not used).

Still, the test started taking a long time. When I debugged it, I noticed this function that was creating the log files. However, the logic was 1) sleep, 2) then do the work, 3) if it fails continue, and then back to 1. This change here does 1) try to do the work, 2) if it fails sleep, then go back to 1, which looks harmless and is probably faster for users and for our tests?